File: 35_최현진/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    special_letters = "~!@#$%^&*()=+[{]}:?,<>/"

    answer = new_id.lower()  # 1단계

    for i in special_letters:
        answer = answer.replace(i, "")  # 2단계

    while '..' in answer:  # 3단계
        answer = answer.replace('..', '.')

    answer = answer.strip('.')  # 4단계

    if answer == "":  # 5단계
        answer = "a"

    if len(answer) > 15:  # 6단계
        answer = answer[:15].strip(".")

    while len(answer) < 3:  # 7단계
        answer += answer[-1]

    return answer

File: 27_이유리/3주차/신규아이디추천.java
class Solution {
    public String solution(String new_id) {
        String answer = "";
        char c;
        
        answer = new_id;
        
        // 1단계
        answer = answer.toLowerCase();
        
        // 2단계
        answer = answer.replaceAll("[^a-z0-9-_.]", "");	// 정규식을 사용하여 소문자 숫자 - _ .을 제외한 모든 문자들을 제거한다.
        
        // 3단계
        answer = answer.replaceAll("[.]{2,}", ".");	// 정규식을 사용하여 마침표가 2번 이상 반복되는 경우 마침표 하나로 대체한다.
        
        // 4단계
        answer = answer.replaceAll("^[.]|[.]$", "");	// 정규식을 사용하여 첫 문자 혹은 끝 문자가 마침표인 경우 제거한다.
        
        // 5단계
        if(answer.equals(""))
        	answer = "a";
        
        // 6단계
        if(answer.length() > 15) {
        	answer = answer.substring(0, 15);
        	answer = answer.replaceAll("[.]$", "");
        }
        
        // 7단계
        if(answer.length() < 3) {
        	c = answer.charAt(answer.length() - 1);
        	while(answer.length() < 3) {
        		answer += c;
        	}
        }
        return answer;
    }
}
